mapiproxy/servers/default/emsmdb/emsmdbp_object.c

```diff
@@ -2631,7 +2631,10 @@ static int emsmdbp_object_get_properties_mapistore_root(TALLOC_CTX *mem_ctx, str
 			}
 		}
                 else {
-			retval = openchangedb_get_folder_property(data_pointers, emsmdbp_ctx->oc_ctx, emsmdbp_ctx->username, properties->aulPropTag[i], folder->folderID, data_pointers + i);
+                        /* We are not using emsmdbp_ctx->username because we want to impersonate to get the properties
+                           on shared folders */
+			owner = emsmdbp_get_owner(object);
+			retval = openchangedb_get_folder_property(data_pointers, emsmdbp_ctx->oc_ctx, owner, properties->aulPropTag[i], folder->folderID, data_pointers + i);
```

Comment:
:hand:  Can you check the behavior of this code with public folders? I have not dig in this further, but I suspect this code may require additional checks to avoid introducing any regression at worst and fix a pending/new bug at best.